pkg/image/installer_image_test.go

```diff
@@ -0,0 +1,152 @@
+package image_test
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/osbuild/images/pkg/container"
+	"github.com/osbuild/images/pkg/customizations/kickstart"
+	"github.com/osbuild/images/pkg/image"
+	"github.com/osbuild/images/pkg/manifest"
+	"github.com/osbuild/images/pkg/ostree"
+	"github.com/osbuild/images/pkg/platform"
+	"github.com/osbuild/images/pkg/rpmmd"
+	"github.com/osbuild/images/pkg/runner"
+	"github.com/stretchr/testify/assert"
+)
+
+func mockPackageSets() map[string][]rpmmd.PackageSpec {
+	return map[string][]rpmmd.PackageSpec{
+		"build": {
+			{
+				Name:     "coreutils",
+				Checksum: "sha256:cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc",
+			},
+		},
+		"os": {
+			{
+				Name:     "kernel",
+				Checksum: "sha256:eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
+			},
+		},
+		"anaconda-tree": {
+			{
+				Name:     "kernel",
+				Checksum: "sha256:eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
+			},
+		},
+	}
+}
+
+func mockContainerSpecs() map[string][]container.Spec {
+	return map[string][]container.Spec{
+		"bootiso-tree": {
+			{
+				Source:  "repo.example.com/container",
+				Digest:  "sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+				ImageID: "sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+			},
+		},
+	}
+}
+
+func mockOSTreeCommitSpecs() map[string][]ostree.CommitSpec {
+	return map[string][]ostree.CommitSpec{
+		"bootiso-tree": {
+			{
+				Ref: "test/ostree/3",
+				URL: "http://localhost:8080/repo",
+			},
+		},
+	}
+}
+
+func TestContainerInstallerUnsetKSOptions(t *testing.T) {
+	img := image.NewAnacondaContainerInstaller(container.SourceSpec{}, "")
+	assert.NotNil(t, img)
+	img.Platform = &platform.X86{
```

Comment:
(niptick) `img.Platform` is set to the same thing in all the tests, maybe worth extracting to make each individual test slightly smaller:
```go
diff --git a/pkg/image/installer_image_test.go b/pkg/image/installer_image_test.
go
index 7fa844806..98310a5c6 100644
--- a/pkg/image/installer_image_test.go
+++ b/pkg/image/installer_image_test.go
@@ -61,29 +61,26 @@ func mockOSTreeCommitSpecs() map[string][]ostree.CommitSpec 
{
        }
 }
 
+var testPlatform = &platform.X86{
+       BasePlatform: platform.BasePlatform{
+               ImageFormat: platform.FORMAT_ISO,
+       },
+       BIOS:       true,
+       UEFIVendor: "test",
+}
+
 func TestContainerInstallerUnsetKSOptions(t *testing.T) {
        img := image.NewAnacondaContainerInstaller(container.SourceSpec{}, "")
        assert.NotNil(t, img)
-       img.Platform = &platform.X86{
-               BasePlatform: platform.BasePlatform{
-                       ImageFormat: platform.FORMAT_ISO,
-               },
-               BIOS:       true,
-               UEFIVendor: "test",
-       }
-       instantiateAndSerialize(t, img, mockPackageSets(), mockContainerSpecs(),
 nil)
+       img.Platform = testPlatform
+       ser := instantiateAndSerialize(t, img, mockPackageSets(), mockContainerS
pecs(), nil)
+       assert.Contains(t, ser, `"inst.ks=hd:LABEL=:/osbuild.ks"`)
 }
 
 func TestContainerInstallerUnsetKSPath(t *testing.T) {
        img := image.NewAnacondaContainerInstaller(container.SourceSpec{}, "")
        assert.NotNil(t, img)
-       img.Platform = &platform.X86{
-               BasePlatform: platform.BasePlatform{
-                       ImageFormat: platform.FORMAT_ISO,
-               },
-               BIOS:       true,
-               UEFIVendor: "test",
-       }
+       img.Platform = testPlatform
        // set empty kickstart options (no path)
        img.Kickstart = &kickstart.Options{}
 
@@ -93,13 +90,7 @@ func TestContainerInstallerUnsetKSPath(t *testing.T) {
 func TestOSTreeInstallerUnsetKSPath(t *testing.T) {
        img := image.NewAnacondaOSTreeInstaller(ostree.SourceSpec{})
        assert.NotNil(t, img)
-       img.Platform = &platform.X86{
-               BasePlatform: platform.BasePlatform{
-                       ImageFormat: platform.FORMAT_ISO,
-               },
-               BIOS:       true,
-               UEFIVendor: "test",
-       }
+       img.Platform = testPlatform
        img.Kickstart = &kickstart.Options{
                // the ostree options must be non-nil
                OSTree: &kickstart.OSTree{},
@@ -111,33 +102,20 @@ func TestOSTreeInstallerUnsetKSPath(t *testing.T) {
 func TestTarInstallerUnsetKSOptions(t *testing.T) {
        img := image.NewAnacondaTarInstaller()
        assert.NotNil(t, img)
-       img.Platform = &platform.X86{
-               BasePlatform: platform.BasePlatform{
-                       ImageFormat: platform.FORMAT_ISO,
-               },
-               BIOS:       true,
-               UEFIVendor: "test",
-       }
-
+       img.Platform = testPlatform
        instantiateAndSerialize(t, img, mockPackageSets(), nil, nil)
 }
 
 func TestTarInstallerUnsetKSPath(t *testing.T) {
        img := image.NewAnacondaTarInstaller()
        assert.NotNil(t, img)
-       img.Platform = &platform.X86{
-               BasePlatform: platform.BasePlatform{
-                       ImageFormat: platform.FORMAT_ISO,
-               },
-               BIOS:       true,
-               UEFIVendor: "test",
-       }
+       img.Platform = testPlatform
        img.Kickstart = &kickstart.Options{}
 
        instantiateAndSerialize(t, img, mockPackageSets(), nil, nil)
 }

```